src/drafts/SelectPanel2/SelectPanel.tsx

```diff
@@ -190,10 +200,14 @@ const SelectPanelHeader: React.FC<React.PropsWithChildren> = ({children, ...prop
           {/* heading element is intentionally hardcoded to h1, it is not customisable 
             see https://github.com/github/primer/issues/2578 for context
           */}
-          <Heading as="h1" sx={{fontSize: 14, fontWeight: 600}} {...props}>
+          <Heading as="h1" id={`${panelId}--title`} sx={{fontSize: 14, fontWeight: 600}} {...props}>
```

Comment:
Seems like we render `Heading` regardless if the title prop is being empty or not. This could lead to label the overlay with an empty string. Should we have a default title as we have have now in SelectPanel v1 or do we need a different enforcement here?

Reply:
> do we need a different enforcement here?

Oh yep! title will be a required prop for SelectPanel